phpBB/phpbb/console/command/user/add.php

```diff
@@ -0,0 +1,318 @@
+	{
+		$this->db = $db;
+		$this->config = $config;
+		$this->language = $language;
+		$this->password_manager = $password_manager;
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->php_ext = $php_ext;
+
+		$this->language->add_lang('ucp');
+		parent::__construct($user);
+	}
+
+	/**
+	 * Sets the command name and description
+	 *
+	 * @return null
+	 */
+	protected function configure()
+	{
+		$this
+			->setName('user:add')
+			->setDescription($this->language->lang('CLI_DESCRIPTION_USER_ADD'))
+			->setHelp($this->language->lang('CLI_HELP_USER_ADD'))
+			->addOption(
+				'username',
+				'U',
+				InputOption::VALUE_REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_USERNAME')
+			)
+			->addOption(
+				'password',
+				'P',
+				InputOption::VALUE_REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_PASSWORD')
+			)
+			->addOption(
+				'email',
+				'E',
+				InputOption::VALUE_REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_EMAIL')
+			)
+			->addOption(
+				'send-email',
+				null,
+				InputOption::VALUE_NONE,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_NOTIFY')
+			)
+		;
+	}
+
+	/**
+	 * Executes the command user:add
+	 *
+	 * Adds a new user to the database. If options are not provided, it will ask for the username, password and email.
+	 * User is added to the registered user group. Language and timezone default to $config settings.
+	 *
+	 * @param InputInterface  $input  The input stream used to get the options
+	 * @param OutputInterface $output The output stream, used to print messages
+	 *
+	 * @return int 0 if all is well, 1 if any errors occurred
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$io = new SymfonyStyle($input, $output);
+
+		$helper = $this->getHelper('question');
+
+		$data = array(
+			'username'     => $input->getOption('username'),
+			'new_password' => $input->getOption('password'),
+			'email'        => $input->getOption('email'),
+		);
+
+		if (!$data['username'])
+		{
+			$question = new Question($this->ask_user('USERNAME'));
+			$data['username'] = $helper->ask($input, $output, $question);
+		}
+
+		if (!$data['new_password'])
+		{
+			$question = new Question($this->ask_user('PASSWORD'));
+			$question->setValidator(function ($value) use ($helper, $input, $output) {
+				$question = new Question($this->ask_user('CONFIRM_PASSWORD'));
+				$question->setHidden(true);
+				if ($helper->ask($input, $output, $question) != $value)
+				{
+					throw new runtime_exception($this->language->lang('NEW_PASSWORD_ERROR'));
+				}
+				return $value;
+			});
+			$question->setHidden(true);
+			$question->setMaxAttempts(5);
+
+			$data['new_password'] = $helper->ask($input, $output, $question);
+		}
+
+		if (!$data['email'])
+		{
+			$question = new Question($this->ask_user('EMAIL_ADDRESS'));
+			$data['email'] = $helper->ask($input, $output, $question);
+		}
```

Comment:
Currently the email is just an option so you add it to the command if you want to send the email. It sounds like what you're asking for would only make sense if the email option is interactive, which currently is not  but it Can be if you prefer.